io/fasta/fasta.go

```diff
@@ -76,6 +80,150 @@ func Parse(r io.Reader) ([]Fasta, error) {
 	return outputFastas, nil
 }
 
+// Parser is a flexible parser that provides ample
+// control over reading fasta-formatted sequences.
+// It is initialized with NewParser.
+
+type Parser struct {
+	// rd keeps state of current reader.
+	rd   bufio.Reader
+	line uint
+}
+
+// NewParser returns a Parser that uses r as the source
+// from which to parse fasta formatted sequences.
+func NewParser(r io.Reader) *Parser {
+	// What's the largest line size a fasta can be?
```

Comment:
I believe there's no upper limit for fasta string length but some do limit to a certain number of characters per line. 70-80 appears to be common.

https://en.wikipedia.org/wiki/FASTA_format#Original_format_&_overview

Reply:
Buffer could be **considerably** smaller then, heh.